recipes/antlr4-cppruntime/all/conanfile.py

```diff
@@ -23,133 +27,113 @@ class Antlr4CppRuntimeConan(ConanFile):
         "shared": False,
         "fPIC": True,
     }
-    settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "pkg_config"
     short_paths = True
 
-    compiler_required_cpp17 = {
+    @property
+    def _minimum_cpp_standard(self):
+        # Antlr 4.9.3 requires C++11 while newer versions require C++17
+        return 17 if Version(self.version) >= "4.10" else 11
+
+    @property
+    def _minimum_compiler_versions_cpp17(self):
+        return {
             "Visual Studio": "16",
+            "msvc": "1923",
```

Comment:
Maybe use [check_min_vs](https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html?highlight=msvc#check-min-vs) to check the Visual Studio / msvc version instead of this lookup. I'm not sure that the `msvc` version will work quite right due to that fourth digit. It would need to be `192` or `193` most likely? Or for some reason maybe `192.3`? There's an example of how to do this in [here](https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/cmake_package/all/conanfile.py#L90), although with how you're using it, you'd need to use a `try...catch` block.

Reply:
Thanks for pointing that out. I have removed both `Visual Studio` and `msvc` from the dictionary. 

`check_min_vs` does not work with `1923` due to the fourth digit, but it turns out that I don't need to check the MSVC version in `_check_minimum_compiler_version_cpp17()`, because the invocation of `check_min_vs(self, "192")` in `validate()` already handles that.